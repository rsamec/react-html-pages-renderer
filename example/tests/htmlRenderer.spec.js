"use strict";
var React = require('react');
var renderer = require('react-test-renderer');
var MobxBinder_1 = require("react-binding/lib/MobxBinder");
var HtmlRenderer_1 = require("../../src/HtmlRenderer");
var WidgetFactory_1 = require("../src/WidgetFactory");
var Freezer = require('freezer-js');
var bindToSchema_1 = require("./../src/utils/bindToSchema");
var personSchema = require("./examples/person.json");
var employeesSchema = require("./examples/employees.json");
describe('Html rendering', function () {
    describe('person', function () {
        //setup
        var frozenSchema = new Freezer(personSchema);
        var dataContext = MobxBinder_1.default.bindToState(personSchema.props.defaultData);
        it('schema renders correctly', function () {
            //setup
            bindToSchema_1.initBindings(frozenSchema, frozenSchema.get(), dataContext);
            var schema = frozenSchema.get();
            //exec
            var component = renderer.create(React.createElement(HtmlRenderer_1.default, { widgets: WidgetFactory_1.default, schema: schema, data: dataContext, pageOptions: schema.props && schema.props.pageOptions }));
            var tree = component.toJSON();
            //verify
            expect(tree).toMatchSnapshot();
        });
    });
    describe('employees', function () {
        //setup
        var frozenSchema = new Freezer(employeesSchema);
        var dataContext = MobxBinder_1.default.bindToState(employeesSchema.props.defaultData);
        it('schema renders correctly', function () {
            //setup
            bindToSchema_1.initBindings(frozenSchema, frozenSchema.get(), dataContext);
            var schema = frozenSchema.get();
            //exec
            var component = renderer.create(React.createElement(HtmlRenderer_1.default, { widgets: WidgetFactory_1.default, schema: schema, data: dataContext, pageOptions: schema.props && schema.props.pageOptions }));
            var tree = component.toJSON();
            //verify
            expect(tree).toMatchSnapshot();
        });
    });
});
