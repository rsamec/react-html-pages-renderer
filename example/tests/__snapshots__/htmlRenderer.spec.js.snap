exports[`Html rendering employees schema renders correctly 1`] = `
<div
  className="printable"
  id="section-to-print"
  style={undefined}>
  <div
    style={Object {}}>
    <div
      style={
        Object {
          "height": undefined,
          "left": undefined,
          "position": "relative",
          "top": undefined,
          "width": undefined,
        }
      }>
      <div>
        <div>
          <div
            style={
              Object {
                "height": undefined,
                "left": 0,
                "position": "relative",
                "top": 0,
                "width": undefined,
              }
            }>
            <div
              style={
                Object {
                  "position": "absolute",
                }
              }>
              <div>
                <span
                  style={Object {}}>
                  1
                </span>
              </div>
            </div>
            <div
              style={
                Object {
                  "position": "absolute",
                }
              }>
              <div>
                <span>
                  Converting circular structure to JSON
                </span>
              </div>
            </div>
            <div
              style={
                Object {
                  "position": "absolute",
                }
              }>
              <div>
                <div
                  style={Object {}}>
                  <div
                    aria-disabled={false}
                    className="InputRange"
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onTouchStart={[Function]}>
                    <span
                      className="InputRange-label InputRange-label--min">
                      <span
                        className="InputRange-labelContainer">
                        0
                      </span>
                    </span>
                    <div
                      className="InputRange-track InputRange-track--container"
                      onMouseDown={[Function]}
                      onTouchStart={[Function]}>
                      <div
                        className="InputRange-track InputRange-track--active"
                        style={
                          Object {
                            "left": "0%",
                            "width": "1%",
                          }
                        } />
                      <span
                        className="InputRange-sliderContainer"
                        style={
                          Object {
                            "left": "1%",
                            "position": "absolute",
                          }
                        }>
                        <span
                          className="InputRange-label InputRange-label--value">
                          <span
                            className="InputRange-labelContainer">
                            1
                          </span>
                        </span>
                        <a
                          aria-controls={undefined}
                          aria-labelledby={undefined}
                          aria-valuemax={100}
                          aria-valuemin={0}
                          aria-valuenow={1}
                          className="InputRange-slider"
                          draggable="false"
                          href="#"
                          onClick={[Function]}
                          onKeyDown={[Function]}
                          onMouseDown={[Function]}
                          onTouchStart={[Function]}
                          role="slider" />
                      </span>
                    </div>
                    <span
                      className="InputRange-label InputRange-label--max">
                      <span
                        className="InputRange-labelContainer">
                        100
                      </span>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            style={
              Object {
                "height": undefined,
                "left": 0,
                "position": "relative",
                "top": 0,
                "width": undefined,
              }
            }>
            <div>
              <div>
                <div
                  style={
                    Object {
                      "height": undefined,
                      "left": 0,
                      "position": "relative",
                      "top": 0,
                      "width": undefined,
                    }
                  }>
                  <div>
                    <div>
                      <div
                        style={
                          Object {
                            "height": undefined,
                            "left": 0,
                            "position": "relative",
                            "top": 0,
                            "width": undefined,
                          }
                        }>
                        <div
                          style={
                            Object {
                              "position": "absolute",
                            }
                          }>
                          <div>
                            <label>
                              <input
                                defaultValue="Samec"
                                onChange={[Function]}
                                placeholder={undefined}
                                type="text"
                                value="Samec" />
                            </label>
                          </div>
                        </div>
                        <div
                          style={
                            Object {
                              "position": "absolute",
                            }
                          }>
                          <div>
                            <label>
                              <input
                                defaultValue="Roman"
                                onChange={[Function]}
                                placeholder={undefined}
                                type="text"
                                value="Roman" />
                            </label>
                          </div>
                        </div>
                        <div
                          style={
                            Object {
                              "position": "absolute",
                            }
                          }>
                          <div>
                            <div
                              style={Object {}}>
                              <div
                                aria-disabled={false}
                                className="InputRange"
                                onKeyDown={[Function]}
                                onKeyUp={[Function]}
                                onMouseDown={[Function]}
                                onTouchStart={[Function]}>
                                <span
                                  className="InputRange-label InputRange-label--min">
                                  <span
                                    className="InputRange-labelContainer">
                                    0
                                  </span>
                                </span>
                                <div
                                  className="InputRange-track InputRange-track--container"
                                  onMouseDown={[Function]}
                                  onTouchStart={[Function]}>
                                  <div
                                    className="InputRange-track InputRange-track--active"
                                    style={
                                      Object {
                                        "left": "0%",
                                        "width": "20%",
                                      }
                                    } />
                                  <span
                                    className="InputRange-sliderContainer"
                                    style={
                                      Object {
                                        "left": "20%",
                                        "position": "absolute",
                                      }
                                    }>
                                    <span
                                      className="InputRange-label InputRange-label--value">
                                      <span
                                        className="InputRange-labelContainer">
                                        2
                                      </span>
                                    </span>
                                    <a
                                      aria-controls={undefined}
                                      aria-labelledby={undefined}
                                      aria-valuemax={10}
                                      aria-valuemin={0}
                                      aria-valuenow={2}
                                      className="InputRange-slider"
                                      draggable="false"
                                      href="#"
                                      onClick={[Function]}
                                      onKeyDown={[Function]}
                                      onMouseDown={[Function]}
                                      onTouchStart={[Function]}
                                      role="slider" />
                                  </span>
                                </div>
                                <span
                                  className="InputRange-label InputRange-label--max">
                                  <span
                                    className="InputRange-labelContainer">
                                    10
                                  </span>
                                </span>
                              </div>
                            </div>
                          </div>
                        </div>
                        <div
                          style={
                            Object {
                              "position": "absolute",
                            }
                          }>
                          <div>
                            <span
                              style={Object {}}>
                              Roman Samec
                            </span>
                          </div>
                        </div>
                      </div>
                    </div>
                    <div>
                      <div
                        style={
                          Object {
                            "height": undefined,
                            "left": 0,
                            "position": "relative",
                            "top": 0,
                            "width": undefined,
                          }
                        }>
                        <div>
                          <div>
                            <div
                              style={
                                Object {
                                  "height": undefined,
                                  "left": 0,
                                  "position": "relative",
                                  "top": 0,
                                  "width": undefined,
                                }
                              }>
                              <div>
                                <div>
                                  <div
                                    style={
                                      Object {
                                        "height": undefined,
                                        "left": 0,
                                        "position": "relative",
                                        "top": 0,
                                        "width": undefined,
                                      }
                                    }>
                                    <div
                                      style={
                                        Object {
                                          "position": "absolute",
                                        }
                                      }>
                                      <div>
                                        <label>
                                          <input
                                            defaultValue={undefined}
                                            onChange={[Function]}
                                            placeholder={undefined}
                                            type="text"
                                            value={undefined} />
                                        </label>
                                      </div>
                                    </div>
                                    <div
                                      style={
                                        Object {
                                          "position": "absolute",
                                        }
                                      }>
                                      <div>
                                        <span
                                          style={Object {}}>
                                          type your content
                                        </span>
                                      </div>
                                    </div>
                                  </div>
                                </div>
                              </div>
                            </div>
                          </div>
                          <div>
                            <div
                              style={
                                Object {
                                  "height": undefined,
                                  "left": 0,
                                  "position": "relative",
                                  "top": 0,
                                  "width": undefined,
                                }
                              }>
                              <div>
                                <div>
                                  <div
                                    style={
                                      Object {
                                        "height": undefined,
                                        "left": 0,
                                        "position": "relative",
                                        "top": 0,
                                        "width": undefined,
                                      }
                                    }>
                                    <div
                                      style={
                                        Object {
                                          "position": "absolute",
                                        }
                                      }>
                                      <div>
                                        <label>
                                          <input
                                            defaultValue={undefined}
                                            onChange={[Function]}
                                            placeholder={undefined}
                                            type="text"
                                            value={undefined} />
                                        </label>
                                      </div>
                                    </div>
                                    <div
                                      style={
                                        Object {
                                          "position": "absolute",
                                        }
                                      }>
                                      <div>
                                        <span
                                          style={Object {}}>
                                          type your content
                                        </span>
                                      </div>
                                    </div>
                                  </div>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            style={
              Object {
                "height": undefined,
                "left": 0,
                "position": "relative",
                "top": 0,
                "width": undefined,
              }
            }>
            <div
              style={
                Object {
                  "position": "absolute",
                }
              }>
              <div>
                <span
                  style={Object {}}>
                  Employees: Roman Samec,
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Html rendering person schema renders correctly 1`] = `
<div
  className="printable"
  id="section-to-print"
  style={undefined}>
  <div
    style={Object {}}>
    <div
      style={
        Object {
          "height": undefined,
          "left": undefined,
          "position": "relative",
          "top": undefined,
          "width": undefined,
        }
      }>
      <div>
        <div>
          <div
            style={
              Object {
                "height": undefined,
                "left": 0,
                "position": "relative",
                "top": 0,
                "width": undefined,
              }
            }>
            <div
              style={
                Object {
                  "position": "absolute",
                }
              }>
              <div>
                <label>
                  <input
                    defaultValue="Roman"
                    onChange={[Function]}
                    placeholder={undefined}
                    type="text"
                    value="Roman" />
                </label>
              </div>
            </div>
            <div
              style={
                Object {
                  "position": "absolute",
                }
              }>
              <div>
                <span
                  style={Object {}}>
                  Roman
                </span>
              </div>
            </div>
            <div
              style={
                Object {
                  "position": "absolute",
                }
              }>
              <div>
                <label>
                  <input
                    defaultValue="Samec"
                    onChange={[Function]}
                    placeholder={undefined}
                    type="text"
                    value="Samec" />
                </label>
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            style={
              Object {
                "height": undefined,
                "left": 0,
                "position": "relative",
                "top": 0,
                "width": undefined,
              }
            }>
            <div
              style={
                Object {
                  "position": "absolute",
                }
              }>
              <div>
                <span
                  style={Object {}}>
                  Roman Samec
                </span>
              </div>
            </div>
            <div
              style={
                Object {
                  "position": "absolute",
                }
              }>
              <div>
                <span
                  style={Object {}}>
                  Roman Samec
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
